modules/apps/transfer/types/transfer_authorization.go

```diff
@@ -31,28 +32,38 @@ func (a TransferAuthorization) Accept(ctx sdk.Context, msg sdk.Msg) (authz.Accep
 		return authz.AcceptResponse{}, errorsmod.Wrap(ibcerrors.ErrInvalidType, "type mismatch")
 	}
 
-	for index, allocation := range a.Allocations {
-		if !(allocation.SourceChannel == msgTransfer.SourceChannel && allocation.SourcePort == msgTransfer.SourcePort) {
-			continue
-		}
+	var (
+		limitLeft  = sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(math.MaxInt64)))
+		isNegative bool
+	)
 
-		limitLeft, isNegative := allocation.SpendLimit.SafeSub(msgTransfer.Token)
-		if isNegative {
-			return authz.AcceptResponse{}, errorsmod.Wrapf(ibcerrors.ErrInsufficientFunds, "requested amount is more than spend limit")
-		}
-
-		if !isAllowedAddress(ctx, msgTransfer.Receiver, allocation.AllowList) {
-			return authz.AcceptResponse{}, errorsmod.Wrap(ibcerrors.ErrInvalidAddress, "not allowed address for transfer")
-		}
-
-		if limitLeft.IsZero() {
-			a.Allocations = append(a.Allocations[:index], a.Allocations[index+1:]...)
-			if len(a.Allocations) == 0 {
-				return authz.AcceptResponse{Accept: true, Delete: true}, nil
+	for index, allocation := range a.Allocations {
+		if allocation.SourceChannel == msgTransfer.SourceChannel && allocation.SourcePort == msgTransfer.SourcePort {
+			for _, coin := range allocation.SpendLimit {
```

Comment:
SpendLimit is defined as `sdk.Coins` so it can contain multiple coins of various denomination. 

If we are iterating on the SpendLimit coins here and doing `SafeSub(msgTransfer.Token)` below then we are deducting more than we should be, right? Essentially over spending.

It seems like this is not a bug at the moment due to the fact that `allocation.SpendLimit` is not being updated / reassigned with the new `limitLeft` until after the loop exits. This behaviour does seem a little off tho.

I'd like to add a test to verify these thoughts. I can check it out later today.

Reply:
Rough testcase could be something like:

```
{
	"test multiple coins does not overspend",
	func() {
		transferAuthz.Allocations[0].SpendLimit = sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(100)), sdk.NewCoin("test-denom", sdk.NewInt(100)), sdk.NewCoin("test-denom2", sdk.NewInt(100)))
		msgTransfer.Token = sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(50))
	},
	func(res authz.AcceptResponse, err error) {
		suite.Require().NoError(err)

		updatedTransferAuthz, ok := res.Updated.(*types.TransferAuthorization)
		suite.Require().True(ok)

		remainder := updatedTransferAuthz.Allocations[0].SpendLimit.AmountOf(sdk.DefaultBondDenom)
		suite.Require().True(sdk.NewInt(50).Equal(remainder))
	},
},
```

Reply:
Nice catch @damiannolan! That logic holds up to me:
```go
for index, allocation := range a.Allocations {
	if allocation.SourceChannel == msgTransfer.SourceChannel && allocation.SourcePort == msgTransfer.SourcePort {
		// imagine: sdk.Coins{token1, token2, token3}
		for _, coin := range allocation.SpendLimit { 
		    limitLeft = SafeSub(token1.Amount) // 3 times
		}
	}
}
    return
```

Reply:
Good point @damiannolan would a simple check for the message transfer Token suffice before subtracting? Something like this 
```go
if coin.Denom == msgTransfer.Token.Denom {
     limitLeft, isNegative = allocation.SpendLimit.SafeSub(msgTransfer.Token)
}
```